c++/src/kj/async.c++

```diff
@@ -2346,7 +2410,15 @@ ChainPromiseNode::ChainPromiseNode(Own<PromiseNode> innerParam)
   inner->onReady(this);
 }
 
-ChainPromiseNode::~ChainPromiseNode() noexcept(false) {}
+ChainPromiseNode::~ChainPromiseNode() noexcept(false) {
+  cancelDependenciesWithoutStackOverflow();
+}
+
+void ChainPromiseNode::cancelDependenciesInto(Vector<Own<PromiseNode>> &toCancel) {
```

Comment:
This logic is broken but for some reason only Windows has coverage. Investigating.

Reply:
Nope. Found it. It's in the `cancelDependenciesWithoutStackOverflow` implementation.